src/engine/ruleset/integrations/system/decoders/system-auth.yml

```diff
@@ -0,0 +1,163 @@
+name: decoder/system-auth/0
+
+sources:
+  - decoder/syslog/0
+
+metadata:
+  module: system-auth
+  title: system-auth logs
+  description: Decoder for system athenticated action logs.
+  compatibility: This integration was tested with logs from OS like Ubuntu 12.04, Centos 7, and macOS Sierra.
+  versions:
+    - "any"
+  author:
+    name: Wazuh Inc.
+    email: info@wazuh.com
+    date: 2023-05-15
+  references:
+    - https://www.loggly.com/ultimate-guide/linux-logging-basics/
+
+definitions:
+  isAuthProcess: >-
+    process.name=="sshd"
+    OR process.name=="sudo"
+    OR process.name=="groupadd"
+    OR process.name=="useradd"
+    OR process.name=="groupdel"
+    OR process.name=="groupmod"
+    OR process.name=="userdel"
+    OR process.name=="usermod"
+    OR +contains/message/pam_unix
```

Comment:
Is it possible to improve this? Treating a log as system-auth just because somewhere in the message it has that word is dangerous and inefficient. For example, the following linux log could match against this decoder when it should not:

```xml
May 30 20:19:19 202-u20-dev-agent sysmon:<Event>
    <System>
        <Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
        <EventID>1</EventID>
        <Version>5</Version>
        <Level>4</Level>
        <Task>1</Task>
        <Opcode>0</Opcode>
        <Keywords>0x8000000000000000</Keywords>
        <TimeCreated SystemTime="2023-05-30T20:19:19.743175000Z"/>
        <EventRecordID>60630</EventRecordID>
        <Correlation/>
        <Execution ProcessID="6176" ThreadID="6176"/>
        <Channel>Linux-Sysmon/Operational</Channel>
        <Computer>202-u20-dev-agent</Computer>
        <Security UserId="0"/>
    </System>
    <EventData>
        <Data Name="RuleName">-</Data>
        <Data Name="UtcTime">2023-05-30 20:19:19.743</Data>
        <Data Name="ProcessGuid">{924935dc-5a47-6476-718a-e63e61550000}</Data>
        <Data Name="ProcessId">39783</Data>
        <Data Name="Image">/usr/bin/grep</Data>
        <Data Name="FileVersion">-</Data>
        <Data Name="Description">-</Data>
        <Data Name="Product">-</Data>
        <Data Name="Company">-</Data>
        <Data Name="OriginalFileName">-</Data>
        <Data Name="CommandLine">grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox pam_unix /var/log/syslog</Data>
        <Data Name="CurrentDirectory">/home/vagrant</Data>
        <Data Name="User">-</Data>
        <Data Name="LogonGuid">{924935dc-0000-0000-ffff-ffffffffffff}</Data>
        <Data Name="LogonId">65535</Data>
        <Data Name="TerminalSessionId">5</Data>
        <Data Name="IntegrityLevel">no level</Data>
        <Data Name="Hashes">-</Data>
        <Data Name="ParentProcessGuid">{00000000-0000-0000-0000-000000000000}</Data>
        <Data Name="ParentProcessId">2384</Data>
        <Data Name="ParentImage">-</Data>
        <Data Name="ParentCommandLine">-</Data>
        <Data Name="ParentUser">-</Data>
    </EventData>
</Event>

```

Reply:
Hum, i will think about it. But i disagree. Log has to be collected from one of the linux defined files and has to be manage by syslog decoder. If a log after this two things has pam_unix it should be manage here. We may need more decoding but i think this is the place. Anyway i will review this. Thanks julian.

Reply:
On linux many processes and modules write /var/log/messages and /var/log/auth. The log I shared is generated in the default sysmon configuration for linux, it is syslog compliant as well as sshd and they write to the same file, so it will be evaluated here. I think these logs should be differentiated by the syslog TAG (process name).
Do we have a case where `+contains/message/pam_unix` is true but the above cases fail? I can't find a test for this case.

Reply:
Hi, we added CRON as part of the expected process list. This allows us to remove the comparison in the check section. 

Hope this helps.